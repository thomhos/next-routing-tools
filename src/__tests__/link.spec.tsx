import React from 'react'
import { create } from 'react-test-renderer'

import { Link, route, RoutesMap, RouteProvider } from '../'

const renderWithContext = (routes: RoutesMap) => (Comp: JSX.Element) =>
    create(<RouteProvider routes={routes}>{Comp}</RouteProvider>)

describe('Link', () => {
    it('has the correct properties', () => {
        expect(Link.contextTypes).toHaveProperty('routes')
    })

    it('renders a basic route', () => {
        const routes = { index: route('/', '/') }
        const link = renderWithContext(routes)(
            <Link href={routes.index.path}>
                <a>Test</a>
            </Link>,
        )

        const snap = link.toJSON()
        expect(snap).toMatchSnapshot()
    })

    it('renders a dynamic route', () => {
        const routes = { blogPost: route('/blog/:id', '/blog') }
        const link = renderWithContext(routes)(
            <Link href="/blog/thing">
                <a>Test</a>
            </Link>,
        )

        const snap = link.toJSON()
        expect(snap).toMatchSnapshot()
    })

    it('renders a multi dynamic route', () => {
        const routes = { blogPost: route('/blog/:id/:action', '/blog') }
        const link = renderWithContext(routes)(
            <Link href="/blog/thing/comment">
                <a>Test</a>
            </Link>,
        )

        const snap = link.toJSON()
        expect(snap).toMatchSnapshot()
    })

    it('renders a dynamic route with query', () => {
        const routes = { blogPost: route('/blog/:id', '/blog') }
        const link = renderWithContext(routes)(
            <Link href="/blog/thing?foo=bar">
                <a>Test</a>
            </Link>,
        )

        const snap = link.toJSON()
        expect(snap).toMatchSnapshot()
    })

    it('renders an external route', () => {
        const routes = { blogPost: route('/blog/:id', '/blog') }
        const link = renderWithContext(routes)(
            <Link href="/external">
                <a>Test</a>
            </Link>,
        )

        const snap = link.toJSON()
        expect(snap).toMatchSnapshot()
    })

    it('renders an external route with query', () => {
        const routes = { blogPost: route('/blog/:id', '/blog') }
        const link = renderWithContext(routes)(
            <Link href="/external?foo=bar">
                <a>Test</a>
            </Link>,
        )

        const snap = link.toJSON()
        expect(snap).toMatchSnapshot()
    })
})
