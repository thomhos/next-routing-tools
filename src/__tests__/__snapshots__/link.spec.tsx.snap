// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Link renders a basic route 1`] = `
<a
  href="/"
  onClick={[Function]}
>
  Test
</a>
`;

exports[`Link renders a dynamic route 1`] = `
<a
  href="/blog/thing"
  onClick={[Function]}
>
  Test
</a>
`;

exports[`Link renders a dynamic route with query 1`] = `
<a
  href="/blog/thing?foo=bar"
  onClick={[Function]}
>
  Test
</a>
`;

exports[`Link renders a multi dynamic route 1`] = `
<a
  href="/blog/thing/comment"
  onClick={[Function]}
>
  Test
</a>
`;

exports[`Link renders an external route 1`] = `
<a
  href="/external"
  onClick={[Function]}
>
  Test
</a>
`;

exports[`Link renders an external route with query 1`] = `
<a
  href="/external?foo=bar"
  onClick={[Function]}
>
  Test
</a>
`;
